idaes/models_extra/power_generation/unit_models/soc_submodels/common.py

```diff
@@ -243,8 +241,59 @@ def _interpolate_2D(
         return (phi_func(icd) - phi_func(icu)) / (cd - cu)
 
 
+class _SubsetOf(object):
+    # Copied and modified from Pyomo's implementation of In
+    #  ___________________________________________________________________________
+    #
+    #  Pyomo: Python Optimization Modeling Objects
+    #  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+    #  Under the terms of Contract DE-NA0003525 with National Technology and
+    #  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+    #  rights in this software.
+    #  This software is distributed under the 3-clause BSD License.
+    #  ___________________________________________________________________________
+    #
+    #  This module was originally developed as part of the PyUtilib project
+    #  Copyright (c) 2008 Sandia Corporation.
+    #  This software is distributed under the BSD License.
+    #  Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+    #  the U.S. Government retains certain rights in this software.
+    #  ___________________________________________________________________________
+    """_SubsetOf(domain, cast=None)
+    Domain validation class admitting a Container of possible values
+
+    This will admit any iterable containing values that are in the
+     `domain` Container (i.e., Container.__contains__() returns True).
+    Most common domains are list, set, and dict objects.
+
+    Parameters
+    ----------
+    domain: Container
+        The container that specifies the allowable values.  Incoming
+        values are passed to ``domain.__contains__()``, and if ``True``
+        is returned, the value is accepted and returned.
+    """
+    #  TODO Need to determine what to do about repeated entries before this can become a Pyomo PR
+    def __new__(cls, domain=None, cast=None):
+        return super(_SubsetOf, cls).__new__(cls)
```

Comment:
This section was derived from `In` in Pyomo, so that's why it's currently that way. Since I hope to ultimately make this a Pyomo PR and then get rid of the implementation here, we can sort out that detail later.